File: javascript/2465.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var distinctAverages = function(nums) {
    nums = nums.sort((a, b) => a - b);
    let average = new Set();
    while(nums.length > 0){
        let min = nums.shift();
        let max = nums.pop();
        average.add((min + max) / 2);
    }
    return average.size;
};

File: c++/2465.cpp
#include <vector>
#include <set>
#include <algorithm>
using namespace std;


class Solution {
public:
    int distinctAverages(vector<int>& nums) {
        sort(nums.begin(), nums.end());
        set<double> averages;
        while (!nums.empty()) {
            int min = nums.front();
            nums.erase(nums.begin());
            int max = nums.back();
            nums.pop_back();
            averages.insert((min + max) / 2.0);
        }
        return averages.size();
    }
};